src/aiida/cmdline/commands/cmd_code.py

```diff
@@ -240,29 +241,35 @@ def show(code):
 @options.OVERWRITE()
 @options.SORT()
 @with_dbenv()
-def export(code, output_file, overwrite, sort):
+def export(code, output_file, overwrite, sort, **kwargs):
     """Export code to a yaml file."""
 
-    import yaml
+    other_args = {'sort': sort, **kwargs}
 
-    code_data = {}
-
-    for key in code.Model.model_fields.keys():
-        value = getattr(code, key).label if key == 'computer' else getattr(code, key)
-
-        # If the attribute is not set, for example ``with_mpi`` do not export it, because the YAML won't be valid for
-        # use in ``verdi code create`` since ``None`` is not a valid value on the CLI.
-        if value is not None:
-            code_data[key] = str(value)
+    if output_file is not None:
+        fileformat = output_file.suffix[1:]
+    else:
+        fileformat = 'yml'
+        output_file = pathlib.Path(f'{code.label}@{code.computer.label}.{fileformat}')
 
     try:
-        output_file = generate_validate_output_file(
-            output_file=output_file, entity_label=code.label, overwrite=overwrite, appendix=f'@{code_data["computer"]}'
+        validate_output_filename(
+            output_file=output_file,
+            overwrite=overwrite,
         )
     except (FileExistsError, IsADirectoryError) as exception:
         raise click.BadParameter(str(exception), param_hint='OUTPUT_FILE') from exception
 
-    output_file.write_text(yaml.dump(code_data, sort_keys=sort))
+    # TODO: In principle, output file validation is also done in the `data_export` function. However, I keep the
+    # `validate_output_file_name` function for now, as it is also used in the `Computer` export, and, as `Computer` is
+    # not derived from `Data`, it cannot be exported by `data_export`.
```

Comment:
Fine to keep the validation, but then rework this TODO as just a comment that validation happens in both places for consistency